### ISSUE TITLE
UnboundLocalError: local variable 'dir_name' referenced before assignment - just a FYI

### ISSUE THREAD

**Issue by Nobrumski** [Confuzu/CivitAI_Image_grabber#16]
I had this pop up today and I realized what caused it. I don't know if any other rejections will cause it, but I was using a username which had been changed since I made note of it. A google search found an image linked to the old name, but when you followed the link it took you to the new username. I don't know if this error will pop up for any other reason, but I am guessing the API is rejecting the request due to unknown name.  It does immediately cause it to fail though even if you have other names, guessing due to unable to create directory. I don't know what the api spits out when you try a wrong username, but tonyhs is what I used that was renamed.

Traceback (most recent call last):
  File "E:\Civitai Image Grabber\civit_image_downloader.py", line 634, in main
    results = await asyncio.gather(*tasks)
  File "E:\Civitai Image Grabber\civit_image_downloader.py", line 554, in download_images
    sort_images_by_model_name(dir_name)
UnboundLocalError: local variable 'dir_name' referenced before assignment

**Comment by Confuzu** [comment]
I fixed it. The script now checks whether the username or modelID exists before it starts creating folders and downloading images.

**Comment by Nobrumski** [comment]
I did get the error again this time on a valid username, Ajuro, on the latest version. It popped at the end so something else may have happened.  It looks like different lines, but not sure with the changes if they match up.

2024-04-15 21:50:18,545 - ERROR - An unexpected error occurred: local variable 'dir_name' referenced before assignment
Traceback (most recent call last):
  File "E:\Civitai Image Grabber\civit_image_downloader.py", line 684, in main
    results = await asyncio.gather(*tasks)
  File "E:\Civitai Image Grabber\civit_image_downloader.py", line 604, in download_images
    sort_images_by_model_name(dir_name)
UnboundLocalError: local variable 'dir_name' referenced before assignment

**Comment by Confuzu** [comment]
i tried the username and it ran without error.  i  can't reproduce the error. 

```
URL https://civitai.com/api/v1/images?cursor=28%3A6%3A13%3A4935847&nsfw=X&username=Ajuro already visited. Ending loop.
2 images have no meta data. 


Number of downloaded images: 1778
Number of skipped images: 3

Reasons for skipping images:
- Network error while downloading the image. Please check your internet connection and try again: 3 times
Failed identifiers:
username: 1923784re
username: eirz39j30fd
Image download `completed.
```

**Comment by Nobrumski** [comment]
It seems to happen when I have multiple names, but if I run the names again individually it won't happen again. This just happened for example. I don't know if this is due to the image that causes it already showing in the history so it is skipped the next run. 

2024-04-20 13:06:20,231 - ERROR - unhandled exception during asyncio.run() shutdown
task: <Task finished name='Task-2' coro=<download_images() done, defined at X:\CivitAI_Image_grabber-main\civit_image_downloader.py:508> exception=UnboundLocalError("local variable 'dir_name' referenced before assignment")>
Traceback (most recent call last):
  File "C:\Users\me\AppData\Local\Programs\Python\Python310\lib\asyncio\runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "C:\Users\me\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 649, in run_until_complete
    return future.result()
  File "X:\CivitAI_Image_grabber-main\civit_image_downloader.py", line 684, in main
    results = await asyncio.gather(*tasks)
  File "X:\CivitAI_Image_grabber-main\civit_image_downloader.py", line 604, in download_images
    sort_images_by_model_name(dir_name)
UnboundLocalError: local variable 'dir_name' referenced before assignment

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "X:\CivitAI_Image_grabber-main\civit_image_downloader.py", line 604, in download_images
    sort_images_by_model_name(dir_name)
UnboundLocalError: local variable 'dir_name' referenced before assignment

**Comment by Nobrumski** [comment]
I am starting to wonder if it is caused by old grabs in the JSON from prior to the fix. 

I ran 4 names tonight which were previously run and had the below error.
2024-04-26 19:50:06,250 - ERROR - An unexpected error occurred: local variable 'dir_name' referenced before assignment
Traceback (most recent call last):
  File "X:\CivitAI_Image_grabber-main\civit_image_downloader.py", line 703, in main
    results = await asyncio.gather(*tasks)
  File "X:\CivitAI_Image_grabber-main\civit_image_downloader.py", line 609, in download_images
    sort_images_by_model_name(dir_name)
UnboundLocalError: local variable 'dir_name' referenced before assignment

Rerunning each name individually and then choosing option 1 to do a fresh download the error didn't show up.

**Comment by Nobrumski** [comment]
I did find one that has the error, both as a clean download and with existing. User=Ainzach

It failed with both re-downloading options same error. Actually now that I look at it, it is slight different line items also, but not sure if it is due to the changes of adding the re-download option. 
From the log
2024-05-06 23:43:11,741 - ERROR - An unexpected error occurred: local variable 'dir_name' referenced before assignment
Traceback (most recent call last):
  File "X:\CivitAI_Image_grabber-main\civit_image_downloader.py", line 703, in main
    results = await asyncio.gather(*tasks)
  File "X:\CivitAI_Image_grabber-main\civit_image_downloader.py", line 609, in download_images
    sort_images_by_model_name(dir_name)
UnboundLocalError: local variable 'dir_name' referenced before assignment

but from the command window
Enter timeout value (in seconds):
Invalid timeout value. Using default value of 20 seconds.
Choose image quality (1 for SD, 2 for HD): 2
Allow re-downloading of images already tracked (1 for Yes, 2 for No) [default: 2]: 1
Choose mode (1 for username, 2 for model ID, 3 for Model tag search): 1
Enter usernames (comma-separated): Ainzach
URL https://civitai.com/api/v1/images?username=Ainzach&nsfw=X already visited. Ending loop.
File "X:\CivitAI_Image_grabber-main\civit_image_downloader.py", line 736, in <module>
    asyncio.run(main())
  File "C:\Users\me\AppData\Local\Programs\Python\Python310\lib\asyncio\runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "C:\Users\me\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 649, in run_until_complete
    return future.result()
  File "X:\CivitAI_Image_grabber-main\civit_image_downloader.py", line 703, in main
    results = await asyncio.gather(*tasks)
  File "X:\CivitAI_Image_grabber-main\civit_image_downloader.py", line 609, in download_images
    sort_images_by_model_name(dir_name)
UnboundLocalError: local variable 'dir_name' referenced before assignment

**Comment by Nobrumski** [comment]
I included screenshots in the other thread. Sorry I forgot I had these and didn't mean to spread it out over multiple. Is there any way to increase the logging to see what Civitai API actually kicks back? Didn't know if it was kicking back with a similar error that the script is just reporting as username when it is rejecting for something else. Wasn't sure if they also introduced a max connection to the public API that was causing it also since I know the script seems to switch back and forth on names in batches or at least that is what the terminal shows.

Maybe it is just the API being spotty, but I usually try to run it late at night when I would think it is getting the least amount of activity. I was able to complete that user today by itself, but errored out with others.
